### DISCUSSION TITLE
Is it possible to globally set combined completion item limit for ALL sources, + other settings applied to every source?

### DISCUSSION THREAD

**Discussion by beauwilliams** [hrsh7th/nvim-cmp#1160] [Q&A] [answered]
I would like to set a global setting, never more than 15 items shown in the completion popup.

This includes all completions combined, added together never more than 15 shown. 

I have found that limiting suggestions per completion, often leads to too many completions on the screen. I can set 15 limit for each completion item, but 15+15+15... is greater than 15. I would not like to limit each source to say 5 only to solve the issue.

Also setting the limit for each source individually, seems to be a bit tedious. 

I looked around but can't find a better way than what I am face with now, is there one?

**Comment by beauwilliams** [comment, marked as answer]
I found the solution I was looking for using this builtin nvim option

`vim.opt.pumheight = 10 -- limit completion items`

**Reply by fitrh** [reply]
Yup, I even set `max_item_count` to the value of `pumheight`

```lua
max_item_count = vim.api.nvim_get_option("pumheight")
```
